File: Programmers/Lv1/Lv1_최대공약수와_최소공배수.py
# 유클리드 호제법을 사용하는 것이 효율면에서 좋다.
def GCD(x, y):
    while y:
        x, y = y, x % y

    return x


def LCM(x, y):
    return (x * y) // GCD(x, y)


def solution(n, m):
    return [GCD(n, m), LCM(n, m)]

File: JavaScript/Lv1_최대공약수와 최소공배수.js
function gcd(a, b) {
  const remainder = a % b;
  if (remainder === 0) return b;
  return gcd(b, remainder);
}

function solution(n, m) {
  var answer = [];
  var result = gcd(n, m);
  answer.push(result);
  answer.push((n * m) / result);
  return answer;
}
